### PR TITLE
Create basic auth integration

### PR DESCRIPTION
**Pull request by mapaiva**
### Summary

Create basic authentication integration containing:

- Redirect to `/login` route in case of unauthorized request made to a protected path
- Redirect from `/login` to `/` home in case of user already logged in accessing the  login screen
- Create basic Spotify Client
- Create basic Spotify Client local storage
- Create blank home screen

### Related to

https://github.com/sputnk/fuzzer/issues/16

### CODE REVIEW

File: web/routes.js [hunk 1]
```diff
@@ -4,16 +4,33 @@ import { createHistory } from 'history'
...
+import SpotifyClient from 'web/lib/Spotify/SpotifyClient'
 
 const browserHistory = useRouterHistory(createHistory)({
   basename: `/fuzzer`
 })
+const spotifyClient = new SpotifyClient({
+  clientId: "a35325e445184826a84e510b8232e0eb",
+  clientSecret: "d71f092ba19e4bf18550f2e5dd4a6e45",
+  redirectURI: "http://localhost:8080/login"
+})
```
Comment (alinemorelli) [inline web/routes.js]: I think it should be an env, so we can change uri to production and development.

File: web/routes.js [hunk 2]
```diff
@@ -4,16 +4,33 @@ import { createHistory } from 'history'
...
+import Home from 'web/views/Home'
+
+import SpotifyClient from 'web/lib/Spotify/SpotifyClient'
 
 const browserHistory = useRouterHistory(createHistory)({
   basename: `/fuzzer`
 })
+const spotifyClient = new SpotifyClient({
+  clientId: "a35325e445184826a84e510b8232e0eb",
+  clientSecret: "d71f092ba19e4bf18550f2e5dd4a6e45",
```
Comment (alinemorelli) [inline web/routes.js]: As our repository is open we should not add it here.

File: web/views/Login/Login.js [hunk 3]
```diff
@@ -7,31 +7,69 @@ import './Login.styl'
...
-          <Background />
+class Login extends React.Component {
+  constructor(props) {
+    super(props)
+    this.onClickLogin = this.onClickLogin.bind(this)
+    this.spotifyClient = new SpotifyClient({
+      clientId: "a35325e445184826a84e510b8232e0eb",
+      clientSecret: "d71f092ba19e4bf18550f2e5dd4a6e45",
+      redirectURI: "http://localhost:8080/login"
+    })
```
Comment (alinemorelli) [inline web/views/Login/Login.js]: As we have authentication on route why we need verify it again here?

File: web/routes.js (hunk 2 above)
Comment (nayramartins) [reply web/routes.js]: Maybe it's better take a look at this https://help.github.com/articles/removing-sensitive-data-from-a-repository/

File: web/routes.js (hunk 1 above)
Comment (mapaiva) [reply web/routes.js]: We definitively should. I think help on that, though

File: web/routes.js (hunk 2 above)
Comment (mapaiva) [reply web/routes.js]: Well noted 👍 . I've already reset the client_secret on Spotify app dashboard and got rid of this from our codebase 😉 

https://github.com/sputnk/fuzzer/pull/40/commits/71d0848a5010053f1d8866d506f44efce67dc48e

File: web/views/Login/Login.js (hunk 3 above)
Comment (mapaiva) [reply web/views/Login/Login.js]: I really dunno if it's a good practice to pass components such as `SpotifyClient` as a property to other components or perhaps create a service (singleton) that is imported by other components.

**Pull request merged**